pandas/tests/dtypes/test_inference.py

```diff
@@ -607,6 +635,42 @@ def test_maybe_convert_objects_nullable_integer(self, exp):
 
         tm.assert_extension_array_equal(result, exp)
 
+    @pytest.mark.parametrize(
+        "exp",
+        [
+            IntegerArray(np.array([2, 0], dtype="i8"), np.array([False, True])),
+            IntegerArray(np.array([2, 0], dtype="int64"), np.array([False, True])),
+        ],
+    )
+    def test_maybe_convert_numeric_nullable_integer(self, exp):
+        # GH 40687
+        arr = np.array([2, np.NaN], dtype=object)
+        result = lib.maybe_convert_numeric(arr, set(), convert_to_nullable_integer=True)
+        tm.assert_extension_array_equal(result, exp)
+
+    @pytest.mark.parametrize(
```

Comment:
can you integrate with float tests

Reply:
Integrated with both int and float. It's still worth keeping this one as a simple basic test of functionality.